pytorch_forecasting/models/deepar/__init__.py

```diff
@@ -194,7 +194,7 @@ def construct_input_vector(
     @property
     def target_position(self):
         variables = self.hparams.time_varying_reals_encoder  # todo: support categorical targets
-        pos = variables.index(self.hparams.target)
+        pos = variables.index(self.hparams.target) + len(self.hparams.static_reals)
```

Comment:
This might not yield the right result because the variables are not necessarily directly mapping to the dataset inputs. What will always work is `self.hparams.x_reals.index(self.hparams.target)`.